File: CPP/Kode Program/Aksesoris.cpp
#include <iostream>
#include <string>
#include "PetShop.cpp"

using namespace std;

// Deklarasi kelas Aksesoris yang merupakan turunan dari PetShop
class Aksesoris : public PetShop {
    private:
        string jenis;   // Jenis aksesoris (kalung, mainan, dll.)
        string bahan;   // Bahan aksesoris (plastik, kain, logam, dll.)
        string warna;   // Warna aksesoris

    public:
        // Konstruktor default
        Aksesoris() : PetShop() {
            this->jenis = "";
            this->bahan = "";
            this->warna = "";
        }

        // Konstruktor dengan parameter
        Aksesoris(int id, string nama_produk, double harga_produk, int stok_produk, string jenis, string bahan, string warna)
        : PetShop(id, nama_produk, harga_produk, stok_produk) {
            this->jenis = jenis;
            this->bahan = bahan;
            this->warna = warna;
        }

        // Setter methods
        void set_jenis(string jenis) {
            this->jenis = jenis;
        }

        void set_bahan(string bahan) {
            this->bahan = bahan;
        }

        void set_warna(string warna) {
            this->warna = warna;
        }

        // Getter methods
        string get_jenis() {
            return this->jenis;
        }

        string get_bahan() {
            return this->bahan;
        }

        string get_warna() {
            return this->warna;
        }

        // Destructor
        ~Aksesoris() {}
};

File: PHP/Kode Program/Aksesoris.php
<?php
require_once "PetShop.php";

class Aksesoris extends PetShop {
    private $jenis;
    private $bahan;
    private $warna;

    // Konstruktor
    public function __construct($id = 0, $nama_produk = "", $harga_produk = 0, $stok_produk = 0, $foto_produk = "", $jenis = "", $bahan = "", $warna = "") {
        parent::__construct($id, $nama_produk, $harga_produk, $stok_produk, $foto_produk);
        $this->jenis = $jenis;
        $this->bahan = $bahan;
        $this->warna = $warna;
    }

    // Setter methods
    public function set_jenis($jenis) {
        $this->jenis = $jenis;
    }

    public function set_bahan($bahan) {
        $this->bahan = $bahan;
    }

    public function set_warna($warna) {
        $this->warna = $warna;
    }

    // Getter methods
    public function get_jenis() {
        return $this->jenis;
    }

    public function get_bahan() {
        return $this->bahan;
    }

    public function get_warna() {
        return $this->warna;
    }
}
?>
